apps/jubmoji-quest/src/pages/index.tsx

```diff
@@ -61,35 +61,27 @@ export default function Home() {
         ) : (
           <>
             {allQuests?.map(
-              ({
-                id,
-                name,
-                description,
-                collectionCards,
-                imageLink,
-              }: JubmojiQuest) => {
+              ({ id, name, description, imageLink, powers }: JubmojiQuest) => {
                 const questPageUrl = `/quests/${id}`;
 
-                const collectionCardIndices = collectionCards.map(
-                  (card) => card.index
-                );
-
                 const questImagePath = imageLink;
 
-                const collected = jubmojis.filter((jubmoji) =>
-                  collectionCardIndices.includes(jubmoji.pubKeyIndex)
+                const numPowersCompleted = powers.filter((power) =>
+                  power.collectionCards
```

Comment:
Don't we need to figure out what type of power this is first before this check? Like if it's NUnique, aren't we supposed to be doing a check that the user has at least N of the collectionCards, not that every single one has been collected?